bittensor/axon.py

```diff
@@ -604,6 +622,55 @@ def default_verify(self, synapse: bittensor.Synapse) -> Request:
         self.nonces[endpoint_key] = synapse.dendrite.nonce
 
 
+class RequestWithBody(Request):
+    """
+    A specialized HTTP request wrapper that includes a pre-defined request body.
```

Comment:
Really cool wrapper, so we cant access the same body once the information has been read by the middle-ware asynchronously and must recreate a second request?

Reply:
Yes unfortunately it was impossible to rewrap the same body. The request lifecycle for starlette doesn't allow for this.

Reply:
I see, are there any increases in memory usage due to the repacking of the request?